core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -213,6 +213,11 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
           data.freeCores = data.totalCores
         }
         makeOffers(executorId)
+
+      case MiscellaneousProcessAdded(time: Long,
```

Comment:
If this's used for YARN only, shall we move it to `YarnSchedulerBackend`? @tgravescs @SaurabhChawla100

Reply:
@Ngone51 - The idea here is to create a framework where any ResourceManager can add their Process info through this MiscellaneousProcessAdded . And the code change is made to provide the info related to yarn-am in case deploy mode client for yarn through this framework.  cc @tgravescs

Reply:
Do we have a clear plan to add support for other resource managers?

Reply:
As of now we are not sure about the process that can be added by the other resource manager.But if there is any process that needs to be shown in the executors page, they can reuse the existing code here.

Reply:
I'm not sure what you had in mind for yarn specific code here?  most of this is in common places so the intent was to make it generic where it could be reused in the future if needed.